optimum/exporters/openvino/model_patcher.py

```diff
@@ -3575,6 +3575,299 @@ def __exit__(self, exc_type, exc_value, traceback):
             block.self_attn.forward = block.self_attn._orig_forward
 
 
+class DeepseekPatcher(DecoderModelPatcher):
+    def __enter__(self):
+        super().__enter__()
+        self_attn = {
+            "deepseek_v3": deepseek_v3_attn_forward,
+            "deepseek_v2": deepseek_v2_attn_forward,
+            "deepseek": minicpm3_attn_forward,
+        }
+
+        self_attn_fwd = self_attn.get(self._model.config.model_type)
+        for block in self._model.model.layers:
+            if self_attn_fwd is not None:
+                block.self_attn._orig_forward = block.self_attn.forward
+                block.self_attn.forward = types.MethodType(self_attn_fwd, block.self_attn)
+            if hasattr(block.mlp, "moe_infer"):
+                block.mlp._org_moe_infer = block.mlp.moe_infer
+                block.mlp.moe_infer = types.MethodType(deepseek_moe_infer, block.mlp)
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        super().__exit__(exc_type, exc_value, traceback)
+        for block in self._model.model.layers:
+            block.self_attn.forward = block.self_attn._orig_forward
+            if hasattr(block.mlp, "_orig_moe_infer"):
+                block.mlp.moe_infer = block.mlp._orig_moe_infer
+
+
+def deepseek_v3_attn_forward(
```

Comment:
could you add a link to where the original updated code comes from ?

Reply:
added